src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/telemetry/telemetry.component.ts

```diff
@@ -0,0 +1,199 @@
+import { Component, OnInit } from '@angular/core';
+import { ValidatorFn, Validators } from '@angular/forms';
+import { Router } from '@angular/router';
+
+import { I18n } from '@ngx-translate/i18n-polyfill';
+import * as _ from 'lodash';
+import { BlockUI, NgBlockUI } from 'ng-block-ui';
+import { forkJoin as observableForkJoin } from 'rxjs';
+
+import { MgrModuleService } from '../../../shared/api/mgr-module.service';
+import { TelemetryService } from '../../../shared/api/telemetry.service';
+import { NotificationType } from '../../../shared/enum/notification-type.enum';
+import { CdFormBuilder } from '../../../shared/forms/cd-form-builder';
+import { CdFormGroup } from '../../../shared/forms/cd-form-group';
+import { CdValidators } from '../../../shared/forms/cd-validators';
+import { NotificationService } from '../../../shared/services/notification.service';
+import { TextToDownloadService } from '../../../shared/services/text-to-download.service';
+
+@Component({
+  selector: 'cd-telemetry',
+  templateUrl: './telemetry.component.html',
+  styleUrls: ['./telemetry.component.scss']
+})
+export class TelemetryComponent implements OnInit {
+  @BlockUI()
+  blockUI: NgBlockUI;
+
+  error = false;
+  configForm: CdFormGroup;
+  licenseAgrmt = false;
+  loading = false;
+  moduleEnabled: boolean;
+  options: Object = {};
+  previewForm: CdFormGroup;
+  requiredFields = [
+    'channel_basic',
+    'channel_crash',
+    'channel_device',
+    'channel_ident',
+    'interval',
+    'proxy',
+    'contact',
+    'description'
+  ];
+  report: object = undefined;
+  reportId: number = undefined;
+  step = 1;
+
+  constructor(
+    private formBuilder: CdFormBuilder,
+    private mgrModuleService: MgrModuleService,
+    private notificationService: NotificationService,
+    private router: Router,
+    private telemetryService: TelemetryService,
+    private i18n: I18n,
+    private textToDownloadService: TextToDownloadService
+  ) {}
+
+  ngOnInit() {
+    this.loading = true;
+    const observables = [
+      this.mgrModuleService.getOptions('telemetry'),
+      this.mgrModuleService.getConfig('telemetry')
+    ];
+    observableForkJoin(observables).subscribe(
+      (resp: object) => {
+        this.moduleEnabled = resp[1]['enabled'];
+        this.options = _.pick(resp[0], this.requiredFields);
+        const configs = _.pick(resp[1], this.requiredFields);
+        this.createConfigForm();
+        this.loading = false;
+        this.configForm.setValue(configs);
```

Comment:
If you change the order of these two lines angular has not to rerender on the first value changes.

```suggestion
        this.configForm.setValue(configs);
        this.loading = false;
```